community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -830,6 +842,32 @@ public void prepareForRecovery() throws IOException
         pagedFile.flushAndForce();
     }
 
+    /**
+     * Must be called after all recovered updates have been applied and before doing further work,
+     * be it reading or writing.
+     *
+     * @throws IOException on {@link PageCache} error.
+     */
+    public void completeRecovery() throws IOException
+    {
+        // For the time being there's an issue where update that come in before a crash
+        // may be applied in a different order than when they get recovered.
+        // This may have structural changes be different during recovery than what they were
+        // during normal application. The result of this is that there may be CRASH pointers left
+        // hanging in the tree. A problem with CRASH pointers is that they can only be recognized
+        // while doing recovery, where they are recognized from their generation being STABLE < gen < UNSTABLE,
+        // i.e. recovery creates this gap to be able to distinguish crashed from fixed pointers.
+        // After recovery is completed and the next checkpoint/close happens that gap is closed
+        // and any remaining CRASH pointers cannot be recognized as such anymore.
+        //
+        // The temporary solution right now is to scan through all tree nodes, as performant as possible,
+        // and zero out all CRASH pointers.
+
+        long generation = this.generation;
+        new CrashGenCleaner( pagedFile, bTreeNode, IdSpace.MIN_TREE_NODE_ID,freeList.lastId() + 1,
+                stableGeneration( generation ), unstableGeneration( generation ), monitor ).clean();
```

Comment:
Actually, checkpoint is performed in the event of kernel panic. This is a problem, but lets consider it to be outside the scope of this PR.
A card has been created.